src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -639,7 +639,7 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
                             per_local.insert(local);
                         }
                     }
-                    cx.per_local[IsNotPromotable].insert(local);
+                    cx.per_local[IsNotConst].insert(local);
```

Comment:
I think we should do this for all locals where `if !temps[local].is_promotable()`.

Reply:
What I meant is that we should do this for all locals, not just temps.
That is, don't do it on a case-by-case basis in this `match`, but rather have this after it:
```rust
            if !temps[local].is_promotable() {
                cx.per_local[IsNotConst].insert(local);
            }
```

Reply:
That's a change that I don't want to backport. are you alright with r+ing this PR and I'll address this immediately in a master-only PR?

Reply:
There's a risk of there being more issues we haven't hit due to this not being general enough.
This is not just code cleanup, I think right now this is kind of broken.